gunrock/graph/csr_parallel_iterator.cuh

```diff
@@ -0,0 +1,66 @@
+// ----------------------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------------------
+
+/**
+ * @file
+ * csr_parallel_iterator.cuh
+ *
+ * @brief CSR Graph Data Structure iterator
+ */
+#pragma once
+
+#include <gunrock/util/array_utils.cuh>
+#include <gunrock/graph/csr.cuh>
+#include <gunrock/graph/graph_base.cuh>
+
+namespace gunrock {
+namespace graph {
+
+/**
+ * @brief ParallelIterator iterator for CSR.
+ *
+ * @tparam VertexT Vertex identifier type.
+ * @tparam SizeT Graph size type.
+ * @tparam ValueT Associated value type.
+ * @tparam GraphFlag graph flag
+ */
+template <typename VertexT, typename SizeT, typename ValueT, GraphFlag FLAG>
+struct ParallelIterator<VertexT, SizeT, ValueT, FLAG, HAS_CSR> {
+  using CsrT =
+      graph::Csr<VertexT, SizeT, ValueT, FLAG & graph::HAS_CSR_MASK,
+                 cudaHostRegisterDefault, (FLAG & graph::HAS_CSR) != 0>;
```

Comment:
I think I'm good with this version, but just curious. Is there any chance that we can simplify by having less files (e.g., this file) if this `struct ParallelIterator` lives inside `struct Csr` in csr.cuh? Would doing so make using this iterator more or less cumbersome?

Reply:
I talked about this with @neoblizz and he preferred doing this in a different file. My first version had the iterator code inside the `struct Csr` and it did look a bit simpler when you construct the iterator but `csr.cuh` has a lot of code inside it already. I like a different file as well, but either way is fine with me.

Reply:
Let's go with what you have.